src/Avalonia.Base/Media/TextFormatting/TextLineImpl.cs

```diff
@@ -1288,11 +1288,10 @@ private TextLineMetrics CreateLineMetrics()
                                 height = drawableTextRun.Size.Height;
                             }
 
-                            if (ascent > -drawableTextRun.Baseline)
-                            {
-                                ascent = -drawableTextRun.Baseline;
-                            }
-
+                            var offset = drawableTextRun.Baseline - descent + ascent;
+                          
+                            ascent -= offset;
```

Comment:
This is equivalent to setting `ascent = descent - drawableTextRun.Baseline`, I think? Just to confirm, the previous value of `ascent` is supposed to be discarded?

Reply:
Offset is just there for debugging. Will rewrite those lines.